Tests/test_file_tiff.py

```diff
@@ -512,13 +510,16 @@ def test_tiled_planar_raw(self):
         with Image.open(infile) as im:
             assert_image_equal_tofile(im, "Tests/images/tiff_adobe_deflate.png")
 
-    def test_palette(self):
-        for mode in ["P", "PA"]:
-            outfile = self.tempfile("temp.tif")
+    def test_palette(self, tmp_path):
+        def roundtrip(mode):
+            outfile = str(tmp_path / "temp.tif")
 
             im = hopper(mode)
             im.save(outfile)
 
+            for mode in ["P", "PA"]:
+                roundtrip(mode)
```

Comment:
This isn't right - you'll figure it out in a second of looking, but just to be explicit - this code never calls `roundtrip`, except from within `roundtrip`.

Reply:
Thanks, fixed in https://github.com/python-pillow/Pillow/pull/4462/commits/fbd4ca1ba0a93fb62da10e9aafd7dc1a965da2b0.